### PR TITLE
Added documentation, with change logs managed by towncrier

### PR DESCRIPTION
**Pull request by andy-maier**
For looking at the documentation, run `make builddoc` with this PR checked out and open `build_docs/index.html` in a browser.

A corresponding project on ReadTheDocs has been created, and the first documentation should be built there when this PR is merged: https://zhmc-log-forwarder.readthedocs.io/en/latest/

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+  configuration: docs/conf.py
+
+# If using Sphinx, optionally build your docs in additional formats such as PDF
+# formats:
+#    - pdf
+
+# Python requirements required to build the docs
+python:
```
Comment (anilak11) [inline .readthedocs.yaml:22]: i'm not an expert here, just a doubt ....can we do something like this

```

python:
   install:
     - requirements:
         - requirements-base.txt
         - requirements.txt
         - requirements-develop.txt
```

will it work, I see we repeated the requirements, so checking if we could avoid that

File: docs/_static/_sphinx_javascript_frameworks_compat.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * Copyright Sphinx contributors
+ * Released under the two clause BSD licence
+ */
+
+/**
+ * small helper function to urldecode strings
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent#Decoding_query_parameters_from_a_URL
+ */
+jQuery.urldecode = function(x) {
```
Comment (anilak11) [inline docs/_static/_sphinx_javascript_frameworks_compat.js:12]: minor: 

```
jQuery.urldecode = function(x) {
    return x ? decodeURIComponent(x.replace(/\+/g, ' ')) : x;
};
```

File: docs/_static/_sphinx_javascript_frameworks_compat.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+ * small helper function to urlencode strings
+ */
+jQuery.urlencode = encodeURIComponent;
+
+/**
+ * This function returns the parsed url parameters of the
+ * current request. Multiple values per key are supported,
+ * it will always return arrays of strings for the value parts.
+ */
+jQuery.getQueryParameters = function(s) {
```
Comment (anilak11) [inline docs/_static/_sphinx_javascript_frameworks_compat.js:29]:
```
jQuery.getQueryParameters = function(s = document.location.search) {
    const result = {};
    const parts = s.slice(s.indexOf('?') + 1).split('&');
    
    parts.forEach(part => {
        const [key, value] = part.split('=', 2).map(jQuery.urldecode);
        result[key] ? result[key].push(value) : result[key] = [value];
    });
    
    return result;
};

```

File: .readthedocs.yaml (hunk 1 above)
Comment (andy-maier) [reply .readthedocs.yaml:22]: I double checked the RTD documentation which shows the repetition: https://docs.readthedocs.io/en/stable/config-file/v2.html#requirements-file

File: docs/_static/_sphinx_javascript_frameworks_compat.js (hunk 2 above)
Comment (andy-maier) [reply docs/_static/_sphinx_javascript_frameworks_compat.js:12]: I copied this Javascript file from somewhere and don't think it is worthwhile changing it (and we would need to change it in all repos for consistency).

File: docs/_static/_sphinx_javascript_frameworks_compat.js (hunk 3 above)
Comment (andy-maier) [reply docs/_static/_sphinx_javascript_frameworks_compat.js:29]: I copied this Javascript file from somewhere and don't think it is worthwhile changing it (and we would need to change it in all repos for consistency).

**Pull request merged**